File: backend/src/hono/routes/tests/file_upload_test.ts
import { Hono } from 'hono';
import { fileUploadRoute } from '../file_upload';

const app = new Hono();

app.route('/files', fileUploadRoute);

describe('File Upload Route', () => {
  it('should upload file', async () => {
    const file = new File(['test content'], 'test_file.txt');
    const res = await app.request('/files', {
      method: 'POST',
      body: file,
    });
    expect(res.status).toBe(201);
    const uploadedFiles = await app.request('/files');
    expect(uploadedFiles.json().length).toBe(1);
  });
});

File: lib/features/file_upload/test/file_upload_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:passdriver/features/file_upload/file_upload_provider.dart';

void main() {
  test('file upload flow', () async {
    final fileUploadProvider = FileUploadProvider();
    final file = File('test_file.txt');
    await file.writeAsString('test content');
    await fileUploadProvider.uploadFile(file);
    final uploadedFiles = await fileUploadProvider.getUploadedFiles();
    expect(uploadedFiles.length, 1);
  });
}
